### PR TITLE
Fix to throw an error in MultipartUpload when copying a file to a temporary path fails.

### PR DESCRIPTION
**Pull request by sudopark**
### Goals :soccer:
When uploading a file with MultipartFormData, if the file copy to the temporary path fails, the entire upload process also fails.

### Implementation Details :construction:
I'm using Alamofire very well in my project. Thank you👍👍. Recently, I found the following errors when uploading files with MultipartFormData.

_Cannot read file at file:///private/var/mobile/Containers/Data/Application/041F8DB7-8C7D-4DC8-A976-4F3B9EACE686/tmp/org.alamofire.manager/multipart.form.data/D80342DA-28F4-40A9-913E-CC42161AE546_

The error above occurred when the uploadTask (with: fromFile:) of URLSsession was called to the nonexistent file. To correct the above problem, I modified the build() of MultipartUpload as follows.

```swift
do {
    try multipartFormData.writeEncodedData(to: fileURL)
} catch {
    // Cleanup after attempted write if it fails.
    try? fileManager.removeItem(at: fileURL)

    // WriteEncodedData(to:) can fail for several reasons, such as when a non-existent file url is entered.
    // In this case, I propose a way to throw through errors to ensure that the build step and all upload processes fail.
    throw error
}

```

### Testing Details :mag:
I have added a test to determine if the upload failed when the actual nonexistent file URL was entered.

### CODE REVIEW

File: Tests/UploadTests.swift
Comment (jshier) [inline Tests/UploadTests.swift:522]:
```suggestion
    func testThatUploadingMultipartFormDataWithNonexistentFileThrowsAnError() {
```

This can be made a bit easier to read.

File: Tests/UploadTests.swift
Comment (jshier) [inline Tests/UploadTests.swift:524]:
```suggestion
        let urlString = URL.makeHTTPBinURL(path: "post")
```

We have APIs for generating our URLs which should be used instead of raw `String`s.

File: Tests/UploadTests.swift
Comment (jshier) [inline Tests/UploadTests.swift:527]:
```suggestion
        let expectation = self.expectation(description: "multipart form data upload from nonexistent file should fail")
```

Again, this can be made a bit easier to read.

File: Tests/UploadTests.swift
Comment (jshier) [inline Tests/UploadTests.swift:525]:
```suggestion
        let imageURL = URL(fileURLWithPath: "does_not_exist.jpg")
```

This makes it a bit safer.

File: Tests/UploadTests.swift
Comment (jshier) [inline Tests/UploadTests.swift:540]:
```suggestion
        waitForExpectations(timeout: timeout)
```

`handler` is unnecessary when `nil`.

File: Tests/UploadTests.swift
Comment (jshier) [inline Tests/UploadTests.swift:544]:
```suggestion
        XCTAssertNil(request.uploadable)
```

Since the `uploadable` value isn't being used it can just be checked inline.

**Review by jshier — CHANGES_REQUESTED** [review]
Nice catch! This change is fundamentally correct but I think the test could be a bit clearer. Also, could you install the latest SwiftFormat and run it in the Alamofire directory? (`swiftformat .`) Once this is updated it can be merged and go out in the next release.

**Comment by sudopark** [comment]
@jshier  Done! thank you for the review 👍

**Review by jshier — APPROVED** [review]
Looks good, I'll merge once the tests pass.

**Review by jshier — APPROVED** [review]
👍

**Review by MarcSteven — COMMENTED** [review]
Looks so cool

**Pull request merged**
